Improved LDAP and coproprate proxy configurations documentation (#10906)
geosolutions-it/MapStore2: docs/developer-guide/configuration-files.md

* Make mapstore configuration system live also as a front-end only framework
* Keep the power of customization provided by spring on the back-end

In this document we will describe the configuration files for both the [back-end](#back-end-configuration-files) and the [front-end](#front-end-configurations-files), and how to [externalize](#externalize-configurations) them. Moreover, we will see how to pass some specific configurations as [Java system properties](#java-system-properties).

## Back-end Configuration Files

They are `.properties` files or `.xml` files, and they allow to configure the various parts of the back-end.

> In source code they are located in `java/web/src/main/resources`. When  `MapStore.war` is build they will be under the directory `/WEB-INF/classes`.

Reviewer: ```suggestion
In source code they are located in `java/web/src/main/resources`. In the `MapStore.war` they are under the directory `/WEB-INF/classes`.
```

```sh
java -Ddatadir.location=/path/to/data-dir -jar mapstore.war
```

In servlet containers like Tomcat, you can set Java system properties in the `setenv.sh` file (for Unix-based systems) or `setenv.bat` file (for Windows systems) located in the `bin` directory of the Tomcat installation.

If you are using Tomcat as a service, you can set Java system properties in the service configuration file (for instance `/etc/default/tomcat9` in Ubuntu 20.04 ).

### `datadir.location`

This property allows you to externalize the location of some back-end and front-end configuration files. For more details, refer to the [Externalize Configuration](externalized-configuration.md#externalized-configuration) section.

### Corporate proxy settings

In case of corporate proxy, some services may require the use of an HTTP proxy to access the internet. You can pass the following Java system properties to configure the HTTP proxy:

* `http.proxyHost`: specifies the HTTP proxy host.
* `http.proxyPort`: specifies the HTTP proxy port.
* `http.nonProxyHosts`: specifies a list of hosts that should not use the HTTP proxy.

> For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html).

Reviewer: @offtherailz we should include also the one for java 11
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html
with a not to check it basing on the used JAVA version from the [ones allowed](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/requirements/#war-installation) (with a link)

> For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html).

Author: ```suggestion
For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html).
```

Author: They are the same property, so I'd simply add the link to v 11 instead of 8 for the moment, without any specification.

> For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html).

Reviewer: ```suggestion
For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html) corresponding to the java version in use (see also https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/requirements/).
```

Reviewer: @offtherailz adjust the requirement link please

> For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html) corresponding to the java version in use (see also https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/requirements/).

Reviewer: @offtherailz we should not use direct links and use relative doc paths

Author: ```suggestion
For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html) corresponding to the java version in use (see also [here](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/requirements/)).
```

> For more information, refer to the [official HTTP proxy documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html) corresponding to the java version in use (see also [here](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/requirements/)).

Reviewer: The relative path @offtherailz
